### ISSUE TITLE
Feature request: custom starter projen configuration file when creating projects

### ISSUE THREAD

**Issue by dontirun** [projen/projen#1468]
### Description

A way to specify/generate a custom `.projen.js` file when creating a new project.

### Use Case

1.  While external modules can be helpful for extending functionality, I am currently using external modules to set defaults for projects for my team (Ex. extending `awscdk-app-ts` to `my-teams-awcdk-app-ts`) which they are then allowed to deviate from . I don't like this approach because this approach is inflexible and makes it difficult to take advantage of new `projen` functionality. A custom `projen` config file would allow users to start with those default props and additionally allow for small customizations using the modules public API.

2. My team creates projects and hands them off to external teams. Those external teams do not have access to our internal artifact repository. A private external module makes it extremely difficult to make that handoff. A custom  starter `projen` config file eliminates that issue

### Proposed Solution

A `--with`  flag for `projen new` that allows for specifying a custom config file.

Ex. `npx projen new awscdk-app-ts  --with my-config@1.2.3`

**Comment by mrgrain** [comment]
I like this idea. Not sure how `--with my-public-config@1.2.3` is any easier to manage than `--from my-public-project-type@1.2.3`

**Comment by dontirun** [comment]
Having two deperate flags (naming aside) would be helpful. I think it's conceptually easier to think of as separate features and there shouldn't be anything preventing you from using a starter config + external module at the same time

**Comment by mrgrain** [comment]
Agreed, thanks for clarifying! I think I was mostly wondering about the issue with private vs public packages you referred to in the original post. I can see how it's more work to setup a whole new project type, especially if you really just want to change some defaults. But I don't see how a starter config would get us around the private/public packages issue.

**Comment by dontirun** [comment]
It gets around the issue of needing to set up a private package in the first place for just some defaults.

But some of those defaults can be "complex" actions like extending tasks

Like you said though, it doesn't help the private/public issue if you need to use a private package in the first place
